packages/http-signature-utils/src/utils/verification.ts

```diff
@@ -0,0 +1,120 @@
+import * as crypto from 'crypto'
+import { RequestLike } from 'http-message-signatures'
+import { verifyContentDigest } from 'httpbis-digest-headers'
+import { importJWK } from 'jose'
+import { JWK } from './jwk'
+
+export function validateHttpSigHeaders(request: RequestLike): boolean {
+  const sig = request.headers['signature']
+  const sigInput = request.headers['signature-input'] as string
+
+  const sigInputComponents = getSigInputComponents(sigInput ?? '')
+  if (
+    !sigInputComponents ||
+    !validateSigInputComponents(sigInputComponents, request)
+  )
+    return false
+
+  return (
+    sig && sigInput && typeof sig === 'string' && typeof sigInput === 'string'
+  )
+}
+
+export async function verifySigAndChallenge(
+  clientKey: JWK,
+  request: RequestLike
+): Promise<boolean> {
+  const sig = request.headers['signature'] as string
+  const sigInput = request.headers['signature-input'] as string
+  const challenge = sigInputToChallenge(sigInput, request)
+  if (!challenge) {
+    return false
+  }
+
+  return await verifySig(sig.replace('sig1=', ''), clientKey, challenge)
+}
+
+//exported for tests
+export async function verifySig(
+  sig: string,
+  jwk: JWK,
+  challenge: string
+): Promise<boolean> {
+  const publicKey = (await importJWK(jwk)) as crypto.KeyLike
+  const data = Buffer.from(challenge)
+  return crypto.verify(null, data, publicKey, Buffer.from(sig, 'base64'))
+}
+
+//exported for tests
+export function sigInputToChallenge(
```

Comment:
I vote we just test `validateHttpSigHeaders` + `verifyHttpSig` (:point_up_2:)
```suggestion
function sigInputToChallenge(
```